### PR TITLE
chore: Bump `schemaVersion` to 13

### LINKED ISSUE #889: Extend schema to support new table component

### PR DESCRIPTION
**Pull request by vsgoulart**
Closes #889

- [ ] This PR adds a new `form-js` element or visually changes an existing component.
  * => In that case, we need to ensure we follow up on this, e.g. by [creating an issue in Tasklist](https://github.com/camunda/tasklist/issues/new/choose)

### CODE REVIEW

File: packages/form-json-schema/README.md [hunk 1]
```diff
@@ -46,11 +46,11 @@ npm run test
 
 ## Schema compatibility notice
 
-This schena is currently only compatible with following [form-js](https://github.com/bpmn-io/form-js) schema versions.
+This schema is currently only compatible with following [form-js](https://github.com/bpmn-io/form-js) schema versions.
 
 | JSON schema version | form-js schema version |
 |---|---|
-| >= 0.2  | <= 12 |
+| >= 0.2  | <= 13 |
```
Comment (pinussilvestrus) [inline packages/form-json-schema/README.md:53]: We will need to change this matrix anyway as we follow the form-js release versions now.

File: packages/form-json-schema/src/defs/examples/components.json [hunk 2]
```diff
@@ -143,5 +143,13 @@
         "components": []
       }
     ]
+    ,
+    [
```
Comment (pinussilvestrus) [inline packages/form-json-schema/src/defs/examples/components.json:147]: Is this valid JSON? 😄

File: packages/form-json-schema/test/spec/validation.spec.js [hunk 3]
```diff
@@ -257,6 +257,15 @@ describe('validation', function() {
...
     testForm('defaultValue-no-array');
 
+
+    testForm('columns-not-allowed');
+
+    testForm('columnsExpression-not-allowed');
+
+    testForm('rowCount-not-allowed');
+
+    testForm('rowCount-no-number');
```
Comment (pinussilvestrus) [inline packages/form-json-schema/test/spec/validation.spec.js:267]:
```suggestion
    testForm('columns-not-allowed');
    

    testForm('columnsExpression-not-allowed');
    

    testForm('rowCount-not-allowed');
    

    testForm('rowCount-no-number');
```

File: packages/form-json-schema/src/defs/component.json [hunk 4]
```diff
@@ -227,6 +227,21 @@
     "properties": {
       "$id": "#/component/properties",
       "$ref": "properties.json"
+    },
+    "columns": {
+      "$id": "#/component/columns",
+      "$ref": "columns.json"
+    },
+    "columnsExpression": {
```
Comment (pinussilvestrus) [inline packages/form-json-schema/src/defs/component.json:235]: We could consider making these two exclusive, so only one can exist in the schema.

However for `values`, `valuesKey` and `valuesExpression` we don't have such a rule as we anyway pick one and ignore the others.

Comment (vsgoulart) [reply packages/form-json-schema/src/defs/component.json:235]: The message is not very clear, I don't know if there's a better way of doing than the way I did

image

File: packages/form-json-schema/src/defs/examples/components.json (hunk 2 above)
Comment (vsgoulart) [reply packages/form-json-schema/src/defs/examples/components.json]: it is, but I improved it anyway

File: packages/form-json-schema/src/defs/component.json (hunk 4 above)
Comment (pinussilvestrus) [reply packages/form-json-schema/src/defs/component.json:235]: In the element-templates-json-schema we have a mechanism to provide custom error messages: https://github.com/camunda/element-templates-json-schema/blob/f06ab6b42a32904eff97688294bd9ad4a994ffe3/packages/zeebe-element-templates-json-schema/src/error-messages.json#L29

We'd have the infrastructure already in place in form-js; the task for generating the errors is there. 

However, these messages are rather for our Desktop Modeler tooling. Monaco / VSCode supports the `errorMessage` keyword, but not with string literals, that's why we separate them. What you could try out is using the `errorMessage` keyword directly somewhere in the rule with a static string.

Comment (vsgoulart) [reply packages/form-json-schema/src/defs/component.json:235]: Done

image

Comment (pinussilvestrus) [reply packages/form-json-schema/src/defs/component.json:235]: Nice one!

**Review by pinussilvestrus — APPROVED** [review]

**Pull request merged**